### ISSUE TITLE
Taghelpers intellisense broken after last update

### ISSUE THREAD

**Issue by ViperGTI007** [aspnet/Razor#2379]
Hi

I updated my Visual Studio on Friday together with the Microsoft.AspNetCore packages (from 2.0.4 to 2.1.0) for my project.  It seems that the upgrade of Microsoft.AspNetCore to 2.1.0 essentially broke my intellisense for TagHelpers.  The behaviour is basically the same as the issue in the link below, except this seems to be related to the Nuget package version for the project rather than Visual Studio itself.
https://github.com/aspnet/Razor/issues/1628

I followed the advice in the link above (when I thought the issue was VS related):  Tried disabling and enabling the Razor services, did a full repair and still nothing.  Both custom developed tag helpers and the standard tag helpers doesn't work with intellisense, although when running the project, they do what they're meant to do.  Everything was working fine on Thursday.

To reproduce, simply open an existing project that uses the Microsoft.AspNetCore v2.0.4 packages, open Nuget and update all packages to the latest versions.  Open any page that uses a TagHelper or try start typing a taghelper on any view.
------------------------------------

Attached are the sample projects to reproduce the issue.  The only difference between the Before and After projects is that I went into Nuget under the Updates tab and chose to update all packages to latest version.
[AspNetCore_TagHelpers_Upgrade.zip](https://github.com/aspnet/Razor/files/2071145/AspNetCore_TagHelpers_Upgrade.zip)

**Comment by orionstudt** [comment]
I can second that my TagHelpers are not working. VS 15.7.2. Project is AspNetCore 2.1.0

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @ViperGTI007, @orionstudt.
Can you please provide us with a repro project, so we can investigate this issue?

**Comment by NTaylorMullen** [comment]
Ya, will need a repro app. I'm not seeing this on my end:
![image](https://user-images.githubusercontent.com/2008729/40931951-bb52be0a-67e1-11e8-9984-bdfa78cef1c2.png)

**Comment by orionstudt** [comment]
Alright, some clarity then on what my specific issue is, maybe. I assumed @ViperGTI007 was having the same issue as me because of his statement:

> essentially broke my intellisense for TagHelpers

I can get you a repro project but I was actually just able to reproduce the issue with a new 2.1.0 MVC template, no auth. Within a razor file it recognizes the `Html` TagHelper, and I can see each method, but I do not get any intellisense for parameters for method completion or any of its overloads.

![image](https://user-images.githubusercontent.com/32800478/40932494-82e7a7e0-67fc-11e8-85a6-9f7c5529353b.png)

**Comment by ViperGTI007** [comment]
I will provide a repo project tomorrow morning.  Doesn't @Html fall under extensions rather than tag helpers?
See below what I mean with tag helpers.  After the upgrade, the green changes to standard blue (like normal html tags) with no intellisense.
![taghelpers-working](https://user-images.githubusercontent.com/25210590/40936863-ff144676-683c-11e8-82a5-bd64057d365b.png)

**Comment by NTaylorMullen** [comment]
> See below what I mean with tag helpers. After the upgrade, the green changes to standard blue (like normal html tags) with no intellisense.

Does your project build after upgrade? If your project is in an invalid state it can impact TagHelper intellisense.

**Comment by orionstudt** [comment]
Alright, thank you!

Sorry I hijacked your post @ViperGTI007!

**Comment by ViperGTI007** [comment]
@NTaylorMullen Yes, it builds and runs correctly.  Just the intellisense is gone.  If you keep typing your tag-helpers and open the page, the tag helpers get generated correctly in the final html.  I've reproduced it with a blank project earlier today, but now need to find it first.
Will create one in the morning quickly.

**Comment by ViperGTI007** [comment]
Sorry for the delay.  I updated the main post with the attachment (repro projects).

**Comment by NTaylorMullen** [comment]
@ViperGTI007 love the repro format you provided 👍

I went ahead and tried out just the "After" and going from "Before" => Update all packages and both resulted in the View having TagHelpers:
![image](https://user-images.githubusercontent.com/2008729/40994218-87376e84-68b0-11e8-98f9-5d0ae6ce2568.png)

My next guess would be that the dotnet CLIs you have installed on your box are funky. Could you do the following:

1. Open VS
2. Open package manager command console
3. Run `dotnet --info`
4. paste the result here

**Comment by ViperGTI007** [comment]
Hi
Thanks.  Here are the results.
PM> dotnet --info
.NET Command Line Tools (2.1.201)

Product Information:
 Version:            2.1.201
 Commit SHA-1 hash:  7932dc6179

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.201\


Microsoft .NET Core Shared Framework Host



  Version  : 2.0.7

  Build    : 2d61d0b043915bc948ebf98836fefe9ba942be11


The Razor Language Services is version 15.7.31476.

As I mentioned in my original post, I originally thought it was my VS installation based on the other issue that I provided the link for so I tried the solutions from that issue which were to Disable and Enable the Razor Language Services (restarting VS between each step obviously) and I did a Repair from the VS installer.  I'm open to suggestions.
Haven't had the time to actually reinstall the whole Visual Studio though, but hopefully that won't be needed.
